client/src/templates/Challenges/utils/worker-executor.js

```diff
@@ -129,16 +129,23 @@ const eventify = self => {
 
   self.emit = (event, ...args) => {
     if (typeof self._events[event] !== 'undefined') {
-      self._events[event].forEach(listener => {
+      let length = self._events[event].length;
+      for (let i = 0; i < length; i++) {
+        const listener = self._events[event][i];
         listener.apply(self, args);
-      });
+        // may remove Listener so should change index and length
+        if (self._events[event].length < length) {
+          i--;
+          length--;
+        }
+      }
```

Comment:
```suggestion
      }
```
Completes above suggestion.